### PR TITLE
EF-52: Ensure submitted field is not empty

### PR DESCRIPTION
**Pull request by olayiwola-compucorp**
## Overview
In this PR we resolve the issue with an error being displayed when clicking `Save Draft` on the Preview page of webforms

## Before
aaaaa

## After
aaaa

## Technical Details
This error happens because the `$form_state['input']['submitted']` array is NULL when the formed is saved as a draft, and RecursiveArrayIterator class expects an array. https://github.com/compucorp/webform_civicrm_membership_extras/blob/162210f2d86c4c83e30643b2cfc97b45a116d882/webform_civicrm_membership_extras.module#L903

To resolve the error we added a check at the top of the function, so it returns immediately if the form is being saved as a draft i.e. `$form_state['input']['submitted']` is empty.

### CODE REVIEW

**Review by omarabuhussein — APPROVED** [review]

**Comment by omarabuhussein** [comment]
@olayiwola-compucorp please don't merge this even if pass testing, I am facing similar issue with normal webform submissions as part of "HCSASPRT-68" ticket, so there might be something else going here that I need to look into.

**Comment by olayiwola-compucorp** [comment]
> @olayiwola-compucorp please don't merge this even if pass testing, I am facing similar issue with normal webform submissions as part of "HCSASPRT-68" ticket, so there might be something else going here that I need to look into.

great chief

**Comment by omarabuhussein** [comment]
Okay, so this seems to "fix" HCSASPRT-68  too, and it seems we will have to go with it for now (once it passes both EF-52 and HCSASPRT-68 testing), but is it a good solution ? Not really!

It seems that `$form_state['input']['submitted']` is not necessarily will be empty only for draft submissions, but sometimes it is also  the case for normal webform submissions on certain webform configurations (I couldn't pin point these configurations yet).

The issue seems to stem from webforms module unreliable APIs (https://www.drupal.org/project/webform/issues/2136771#comment-8190157), in which there is no proper way to get the existing webform submitted values, and combined with CiviCRM webform_civicrm customizations on top of it, things become more harder. 

I also found issues with pro-rating not working when the membership type is selected directly on the civicrm tab inside the webform admin page, that is because `_webform_civicrm_membership_extras_get_membership_types` function tries to look the membership type in the webform `components` , but the membership type in such case is not part of the components, and rather stored in the `webform_civicrm` object.

For proper fix for this issue, we want to revise the whole pro-rata and discount functionality inside this module (which seems to be very messy anyway), and properly rewrite it (or some parts of it) so it can handle different webforms configs such as:

 - Webforms where membership is selected by the admin on the civicrm tab.
 - Webforms where the user can select the membership on the webform.
 - Cases where the membership dates can be selected manually by the user (more on this down).
 - Cases where there are memberships under nested elements.

Some of the stuff to be tested once we do the above includes what can be found in these PRs

- https://github.com/compucorp/webform_civicrm_membership_extras/pull/36
- https://github.com/compucorp/webform_civicrm_membership_extras/pull/38
- https://github.com/compucorp/webform_civicrm_membership_extras/pull/41
- https://github.com/compucorp/webform_civicrm_membership_extras/pull/42
- https://github.com/compucorp/webform_civicrm_membership_extras/pull/45


Now the reason that I am okay with this fix for now, because it happens in a function that tries to get the membership dates, in case they where selected manually by the user, which as it seems to me, was only added to help QA in their testing, thus we don't officially support pro-rate for manual date selection on webforms yet. thus having that functionally broken is fine for the time being, until we a proper solution.

**Pull request merged**
